### ISSUE TITLE
show news diff (especially breaking changes) on startup

### ISSUE THREAD

**Issue by justinmk** [neovim/neovim#21431]
### Feature description

# Problem

Currently people on development branch (master) must follow https://github.com/neovim/neovim/issues/14090 or manually read `:help news` to see recent changes.

# Proposal

1. On startup, asynchronously (CursorHold?), store the contents of `news.txt` at `sdtpath('state')`
    - also store the current Nvim version in a shada-stored variable like `g:NVIM_VERSION`
2. On startup, compare the current Nvim version to `g:NVIM_VERSION`.
    - related: https://github.com/neovim/neovim/issues/20981
3. If current version is newer, store it in `g:NVIM_VERSION`.
4. Diff the old news.txt with the current news.txt. If it contains breaking changes, show a modal prompt like:
    - `There breaking changes since the previously used version of Nvim x.y.z-123, show the diff of news.txt?`

**Comment by XaurDesu** [comment]
So, would i be able to help around on this issue?

**Comment by lewis6991** [comment]
I would find this a little annoying. Could we make it opt-in/out?

**Comment by clason** [comment]
Agree; I also worry about the (performance) implications of the necessary startup shenanigans.

We already have a friendly
```
 type  :help news<Enter> to see changes in v0.9
```

in the intro screen. Of course, we could make that blink red the first time a new version is started 😬

**Comment by XaurDesu** [comment]
@lewis6991 do you think it would be better for it to be opt-in or opt-out? I'd probably imagine that it would mostly seem a better idea to make it an opt-in program, mostly because i'd imagine the people this would help would probably already know that they need to see `news.txt,` at startup and that probably would decide on their own accord for it to be worth it. thing that wouldn't be the case for everyone (i mean, you don't seem like someone that would use it, for example) but i'm pretty new around here, so i'd rather ask what do you guys think is the best course of action on these changes. @clason also seems to bring an interesting point as the proposal would probably be an easier alternative, anyways.

**Comment by clason** [comment]
(I was kidding, in case this wasn't obvious.)

**Comment by XaurDesu** [comment]
sorry about that, then.

**Comment by justinmk** [comment]
> in the intro screen. Of course, we could make that blink red the first time a new version is started 😬

:) That's not a bad idea. In either case, part of the requirements is to find a way to avoid an over-frequent notification. Core contributors will notice this (since they are rebuilding Nvim a lot) so I don't think that's a big risk.
